skglm/estimators.py

```diff
@@ -971,8 +1179,61 @@ def path(self, X, y, alphas, coef_init=None, return_n_iter=True, **params):
             return_n_iter=return_n_iter, max_epochs=self.max_epochs,
             p0=self.p0, tol=self.tol, verbose=self.verbose)
 
+    def predict_proba(self, X):
+        """Probability estimates.
+
+        The returned estimates for all classes are ordered by the
+        label of classes.
+        For a multi_class problem, a one-vs-rest approach, i.e calculate the probability
+        of each class assuming it to be positive using the logistic function.
+        and normalize these values across all the classes.
+
+        Parameters
+        ----------
+        X : array-like of shape (n_samples, n_features)
+            Vector to be scored, where `n_samples` is the number of samples and
+            `n_features` is the number of features.
+
+        Returns
+        -------
+        T : array-like of shape (n_samples, n_classes)
+            Returns the probability of the sample for each class in the model,
+            where classes are ordered as they are in ``self.classes_``.
+        """
+        check_is_fitted(self)
+        if len(self.classes_) > 2:
+            # Code taken from https://github.com/scikit-learn/scikit-learn/
+            # blob/c900ad385cecf0063ddd2d78883b0ea0c99cd835/sklearn/
+            # linear_model/_base.py#L458
+            def _predict_proba_lr(X):
+                """Probability estimation for OvR logistic regression.
+
+                Positive class probabilities are computed as
+                1. / (1. + np.exp(-self.decision_function(X)));
+                multiclass is handled by normalizing that over all classes.
+                """
+                prob = self.decision_function(X)
+                expit(prob, out=prob)
+                if prob.ndim == 1:
+                    return np.vstack([1 - prob, prob]).T
+                else:
+                    # OvR normalization, like LibLinear's predict_probability
+                    prob /= prob.sum(axis=1).reshape((prob.shape[0], -1))
+                    return prob
+            # OvR normalization, like LibLinear's
+            return _predict_proba_lr(X)
+        else:
+            decision = self.decision_function(X)
+            if decision.ndim == 1:
+                # Workaround for multi_class="multinomial" and binary outcomes
+                # which requires softmax prediction with only a 1D decision.
+                decision_2d = np.c_[-decision, decision]
+            else:
+                decision_2d = decision
+            return softmax(decision_2d, copy=False)
+
 
-class LinearSVC(LinearSVC_sklearn):
+class LinearSVC(LinearClassifierMixin, SparseCoefMixin, BaseEstimator):
```

Comment:
I will add as a TODO for another PR to add a predict_proba for LinearSVC, probably just taking some code from sklearn.